game/tasks/task01_S_player.py

```diff
@@ -0,0 +1,190 @@
+lets provide a little more background on the Single Responsibility Principle (SRP)
+
+In the above code, there was a class with one function 'pick_resource_path'. You may
+have noticed that this made editing the resource path very straightforward. Regardless
+of if the class implemented random selection or returned a hard coded string, the does
+one thing.
+
+As an alternative to how the above code could have been implemented, we could have
+written everything in a single class like below:
+
+"""
+
+
+class OneClass:
+    """Rewritten version of the above code without SRP."""
+
+    def __init__(self) -> None:
+        self.resource_path = ':characters:Female/Female 02-3.png'
+
+    def make_sprite(self) -> arcade.Sprite:
+        return PlayerSprite(self.resource_path)
+
+    def register_sprite(self, sprite_register: SpriteRegister) -> None:
+        register = Register(sprite=self.make_sprite, source=SOURCE_NAME)
+        sprite_register.register_sprite(register)
+
+
+"""
+
+Now let's say that we decide we want to add the random selection or we want to change
+which type of Sprite is created. We may then end up writing code like this:
+
+"""
+
+
+class OneClassWithExtraLogic:
+    """Above code with logic to select the sprite type."""
+
+    def __init__(self, is_player: bool = True) -> None:
+        self.resource_path = ':characters:Female/Female 02-3.png'
+        self.is_player = is_player
+
+    def make_player_sprite(self) -> arcade.Sprite:
+        return PlayerSprite(self.resource_path)
+
+    def make_game_sprite(self) -> arcade.Sprite:
+        return CharacterSprite(self.resource_path)
+
+    def register_sprite(self, sprite_register: SpriteRegister) -> None:
+        sprite = self.make_player_sprite() if self.is_player else self.make_game_sprite()
+        register = Register(sprite=sprite, source=SOURCE_NAME)
+        sprite_register.register_sprite(register)
+
+
+def load_sprites_1(sprite_register: SpriteRegister) -> None:
+    """Example using the OneClassWithExtraLogic."""
+    OneClassWithExtraLogic(is_player=True).register_sprite(sprite_register)
+
+
+"""
+
+Following SRP, this class could be rewritten to separate responsibilities between
+selecting the sprite and registering.
+
+"""
+
+
+class Sprite2:
+    """Just a Sprite."""
+
+    def __init__(self) -> None:
+        self.resource_path = ':characters:Female/Female 02-3.png'
+
+    def make_player_sprite(self) -> arcade.Sprite:
+        return PlayerSprite(self.resource_path)
+
+    def make_game_sprite(self) -> arcade.Sprite:
+        return CharacterSprite(self.resource_path)
+
+
+class SpriteRegister2:
+    """Just a Register method."""
+
+    def register_sprite(self, sprite: arcade.Sprite, sprite_register: SpriteRegister) -> None:
+        register = Register(sprite=sprite, source=SOURCE_NAME)
+        sprite_register.register_sprite(register)
+
+
+def load_sprites_2(sprite_register: SpriteRegister) -> None:
```

Comment:
What's the purpose of adding numbers at the end of `load_sprite`, `Sprite` and `SpriteRegister`?

Reply:
The numbers are to avoid shadowing/name collision, but I can rework those to be underscores and add a note